backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/UserAuthorizationVerifier.java

```diff
@@ -36,6 +39,17 @@ public boolean userHasSiteAdminRole() {
 
     public boolean userHasPermission(UserPermission permission) {
         Optional<OrganizationRoles> orgRoles = _orgService.getCurrentOrganizationRoles();
-        return orgRoles.isPresent() && orgRoles.get().getGrantedPermissions().contains(permission);
+        // more troubleshooting help here.
+        // Note: if your not reaching this code, then grep for 'AbstractAccessDecisionManager.accessDenied' in
+        // spring library AffirmativeBased.java and set a breakpoint there.
+        if (orgRoles.isEmpty()) {
+            LOG.warn("Permission request for {} failed. No roles for org defined.", permission);
+            return false;
+        }
+        if (!orgRoles.get().getGrantedPermissions().contains(permission)) {
+            LOG.warn("Permissions request for {} failed. Not a granted permission.", permission);
```

Comment:
Interesting point: do we want a warning every time this happens? My first instinct is no, but actually people really shouldn't be hitting endpoints they're not allowed very often absent bugs or attempts at privilege escalation, so `WARN` is probably the right level.